src/transcode.jl

```diff
@@ -27,21 +30,34 @@ julia> String(decompressed)
 
 ```
 """
-function Base.transcode(::Type{C}, data::ByteData) where C<:Codec
+function Base.transcode(::Type{C}, args...) where {C<:Codec}
     codec = C()
     initialize(codec)
     try
-        return transcode(codec, data)
+        return transcode(codec, args...)
     finally
         finalize(codec)
     end
 end
 
+_default_output_buffer(codec, input) = Buffer(
+    initial_output_size(
+        codec,
+        buffermem(input)
+    )
+)
+
 """
-    transcode(codec::Codec, data::Vector{UInt8})::Vector{UInt8}
+    transcode(
+        codec::Codec,
+        data::Union{Vector{UInt8},Base.CodeUnits{UInt8},Buffer},
+        [output::Union{Vector{UInt8},Base.CodeUnits{UInt8},Buffer}],
+    )::Union{Vector{UInt8},Base.CodeUnits{UInt8}}
```

Comment:
This signature line implies we can mix and match `Vector{UInt8}` and `Base.CodeUnits{UInt8}`. Is that true? Do we want that?

Can we document what the output type will be?

Reply:
I see no reason to forbid mix and match types. It's more flexibility for users at no performance cost. I'm not against enforcing the same type either, you tell me.

Regarding the return value, I believe it's always `Vector{UInt8}`, right? Because all these things are wrapped in `Buffer` and then returned as `buffer.data` here [1]. I've documented this.

[1] https://github.com/JuliaIO/TranscodingStreams.jl/blob/e926f4faa7941c979416d195abff81fe52f644b8/src/transcode.jl#L132

Reply:
I consider this done, let me know.